perf/lua/tuple_field_by_name.lua

```diff
@@ -0,0 +1,73 @@
+local benchmark = require('benchmark')
+local clock = require('clock')
+local decimal = require('decimal')
+
+local USAGE = [[
+ This benchmark measures the performance of accessing tuple fields by their
+ names referenced in the tuple format.
+]]
+
+local params = benchmark.argparse(arg, {}, USAGE)
+
+local bench = benchmark.new(params)
+
+local CYCLES = 3e8
+
+local decimal_value = decimal.new(1)
+
+local tuple_formats = {
+    {'field', 'number'},
+    {'field', 'string'},
+    {'field', 'decimal'},
+}
```

Comment:
This definitely requires an explanation. Why these types were chosen? `number` and `string` are scalars, while `decimal` is a MsgPack extension, and it's encoding/decoding is more complicated than for scalars, see tarantool msgpack extensions in [^1].

I see two possible options here:
1. remove decimal type and leave scalars only (because encoding/decoding overhead is quite similar)
2. report numbers for each datatype separately and probably add uuid, datetime etc.

I would choose 1st option and instead add another microbench for encoding/decoding msgpack types (I'm wondering it was not added in scope of https://github.com/tarantool/tarantool/pull/10883).

[^1]: https://www.tarantool.io/en/doc/latest/reference/internals/msgpack_extensions/

Reply:
Different data types already reported separately, missed it
![image](https://github.com/user-attachments/assets/20e4163d-e3d9-4605-a206-67d30d8946d6)

Reply:
I guess that `decimal` here is as result of our discussion, where I wondered about data types that decoded to cdata values. So, the reason is not using the msgpack extension or more complicated encoding format (however that's true that all the tarantool's extension types are decoded to cdata).

I agree that it may be unclear and some remark would be good.

Reply:
> 2\. report numbers for each datatype separately and probably add uuid, datetime etc.

Honestly, it looks more reliable to me, since we may compare their behaviour. What data types exactly do you want to see in the benchmark? All extensions of msgpack (I added this one exactly with these thoughts, but I am not sure that the difference between extensions is vital)?

> I would choose 1st option and instead add another microbench for encoding/decoding msgpack types (I'm wondering it was not added in scope of #10883).

I like the idea of adding the msgpack decoder/encoder benchmarks for both the FFI and Lua C implementations (see #5896, for example). But it stands out of scope of this particular PR. Should I create a relevant ticket?

Reply:
I propose to just add a comment that describes your motivation with decimal. You already described it in #11175:

> The most common cases of tuples containing strings or numbers are accessed 1.5-2.5 times faster. The drawback is the field contains the cdata from the msgpack extensions (like decimals). Their decoding still takes approximately the same time (maybe 5% slower).

i.e. `decimal` is just a one of custom MsgPack datatype with cdata.

Reply:
Added the comment:
```diff
@@ -18,6 +19,8 @@ local decimal_value = decimal.new(1)
 local tuple_formats = {
     {'field', 'number'},
     {'field', 'string'},
+    -- Use the decimal as one of the custom MsgPack datatypes with
+    -- cdata.
     {'field', 'decimal'},
 }
```